Tests/MongoSwiftTests/CrudTests.swift

```diff
@@ -239,7 +231,115 @@ private class AggregateTest: CrudTest {
 
 private class BulkWriteTest: CrudTest {
     override func execute(usingCollection coll: MongoCollection<Document>) throws {
-        XCTFail("Unimplemented")
+        let requestDocuments: [Document] = try self.args.get("requests")
+        let requests = try requestDocuments.map { try BulkWriteTest.parseWriteModel($0) }
+        let options = BulkWriteTest.parseBulkWriteOptions(self.args["options"] as? Document)
+
+        if let result = try coll.bulkWrite(requests, options: options) {
+            verifyBulkWriteResult(result)
+        }
+    }
+
+    private static func parseBulkWriteOptions(_ options: Document?) -> BulkWriteOptions? {
+        guard let options = options else {
+            return nil
+        }
+
+        let ordered = options["ordered"] as? Bool ?? true
+
+        return BulkWriteOptions(ordered: ordered)
+    }
+
+    private typealias DeleteOneModel = MongoCollection<Document>.DeleteOneModel
+    private typealias DeleteManyModel = MongoCollection<Document>.DeleteManyModel
+    private typealias InsertOneModel = MongoCollection<Document>.InsertOneModel
+    private typealias ReplaceOneModel = MongoCollection<Document>.ReplaceOneModel
+    private typealias UpdateOneModel = MongoCollection<Document>.UpdateOneModel
+    private typealias UpdateManyModel = MongoCollection<Document>.UpdateManyModel
```

Comment:
This isn't technically necessary, but it makes the cases below a bit more concise. Copied this from `MongoCollection+BulkWriteTests.swift`.

Reply:
This may depend on the outcome of https://github.com/mongodb/mongo-swift-driver/pull/97#discussion_r216958450, so we can punt for now.

If we do want to share these, am I correct in thinking I'd simply need to make them `internal` at the top-level scope of a test file in order to make them accessible throughout the MongoSwiftTests package?

Reply:
yes, I believe that's correct.